content/config/otel.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Open Telemetry for the Docker CLI
+description: Learn about how you can leverage the Docker's capabilities to capture OTel traces
+keywords: otel, telemetry, traces, metrics, logs
+---
+
+{{< introduced engine 26.1.0 >}}
+
+The Docker Engine supports configurable [OTel](https://opentelemetry.io/docs/)
+utilities for tracing and monitoring. You can use these capabilities to capture
+information about `docker` CLI command invocations to get better insight into
+your Docker usage.
+
+## What is OTel?
+
+OTel stands for "Open Telemetry", and it's an open observability framework for
+creating and managing telemetry data, such as traces, metrics, and logs.
+OpenTelemetry is vendor- and tool-agnostic, meaning that it can be used with a
+broad variety of Observability backends.
+
+Support for OTel in the Docker CLI means that the CLI can emit information
+about events that take place, using the protocols and conventions defined in
+the Open Telemetry specification.
+
+## How it works
+
+The Docker CLI doesn't emit telemetry data by default. Only if you've set an
+environment variable on your system will Docker CLI attempt to emit OTel
+metrics, to the endpoint that you specify.
+
+```bash
+DOCKER_CLI_OTEL_EXPORTER_OTLP_ENDPOINT=<endpoint>
+```
+
+The variable specifies the endpoint of where telemetry data about `docker` CLI
+invocation should be sent. To capture the data, you'll need an OTel collector
+that's listening on that endpoint.
+
+A collector receives the telemetry data, processes it, and exports it to a
+backend. The backend is where the telemetry data gets stored. You can choose
+from a number of different backends, such as Prometheus or InfluxDB.
+
+Some backends provide tools for visualizing the metrics directly.
+Alternatively, you can also run a dedicated frontend, such as Grafana.
+
+## Setup
+
+To get started capturing telemetry data for the Docker CLI, you'll need to:
+
+- Set the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable to point to an OTel collector endpoint
+- Run an OTel collector that receives the signals from CLI command invocations
+- Run a backend for storing the data received from the collector
+
+The following Docker Compose file bootstraps a service to get started with
+OTel. It includes an OTel collector and a Prometheus backend.
+
+```yaml {collapse=true,title=compose.yml}
+name: cli-otel
+services:
+  prometheus:
+    image: prom/prometheus
+    command:
+      - "--config.file=/etc/prometheus/prom.yml"
+    ports:
```

Comment:
Wondering if it would work to mount the config over the default one that's provided; I noticed the image has some default options set, and this would mean that we're currently removing some of the other ones (do we know if that has an effect on what it does? At least it looks like this would mean changing the default storage location);


```bash
docker image inspect --format '{{json .Config.Cmd}}' prom/prometheus | jq .
[
  "--config.file=/etc/prometheus/prometheus.yml",
  "--storage.tsdb.path=/prometheus",
  "--web.console.libraries=/usr/share/prometheus/console_libraries",
  "--web.console.templates=/usr/share/prometheus/consoles"
]
```

Reply:
The idea was more to provide a smallest working example than to provide an exhaustive guide on prometheus, so I removed anything that wasn't essential. Maybe we could mount the new one on top of that so we don't need to specify `--config.file`, but I don't think we should worry about removing any configurations as long as they don't break the setup.